create_release_pr/action.yml

```diff
@@ -0,0 +1,70 @@
+name: Automate creating the Release pull request
+description: Automate creating the Release pull request
+inputs:
+  version:
+    description: Current release version
+    required: true
+  token:
+    description: The GitHub PAT for this action to use
+    required: false
+    default: ${{ github.token }}
+  base_branch:
+    description: The base branch for the release pull request, e.g., "main"
+    required: false
+    default: main
+  changelogConfiguration:
+    description: The changelog configuration file path, e.g., ".github/workflows/config/changelog-release.json"
+    required: true
+  assignee:
+    description: The assignee for the Release pull request, e.g., bot-nimble
+    required: false
+    default: 'bot-nimble'
+  label:
+    description: 'The label for the Release pull request, e.g., "type : release"'
+    required: false
+    default: 'type : release'
+
+runs:
+  using: composite
+  steps:
+    - name: Find HEAD commit
+      id: head
+      shell: bash
+      run: echo "sha=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+
+    - name: Build changelog
+      id: changelog
+      uses: mikepenz/release-changelog-builder-action@v4
+      with:
+        configuration: ${{ inputs.changelogConfiguration }}
+        # Listing PRs from the last tag to the HEAD commit
+        toTag: ${{ steps.head.outputs.sha }}
```

Comment:
On the first release time, we didn't have any tag, and we don't define the fromTag here; it can't detect the changes, when I tested on my PR, the output is

- fromTag: the last commit on the develop branch
- toTag: the last commit on the current branch (also develop)

So it show `--no-changes` in the payload

Then I changed the config by setting the fromTag here; it's able to generate the changelog content. 

- fromTag: 'fixed_last_commit_on_main'
- toTag: ${{ steps.head.outputs.sha }}

For that, should we consider defining the from here?

Reply:
Our final solution looks like this FYI ;) 

```
runs:
  using: composite
  steps:
    - name: Detect latest develop commit
      id: detect_latest_develop_commit
      shell: bash
      run: |
        git fetch origin develop
        echo "sha=$(git rev-parse origin/develop)" >> $GITHUB_OUTPUT

    - name: Detect latest main commit
      id: detect_latest_main_commit
      shell: bash
      run: |
        git fetch origin main
        echo "sha=$(git rev-parse origin/main)" >> $GITHUB_OUTPUT

    - name: Generate changelog
      id: generate_changelog
      uses: mikepenz/release-changelog-builder-action@v4
      with:
        token: ${{ inputs.github_token }}
        configuration: ${{ inputs.changelog_configuration }}
        outputFile: changelog.txt
        fromTag: ${{ steps.detect_latest_main_commit.outputs.sha }}
        toTag: ${{ steps.detect_latest_develop_commit.outputs.sha }}

    - name: Prepend Release URL into changelog
      shell: bash
      run: |
        echo -e "${{ inputs.release_body_url }}\n$(cat changelog.txt)" > changelog.txt

    - name: Create the Release pull request
      env:
        GH_TOKEN: ${{ inputs.github_token }}
      shell: bash
      run: |
        RELEASE_BRANCH=release/${{ inputs.release_version }}

        # Create the release branch
        git checkout -b $RELEASE_BRANCH
        git push origin $RELEASE_BRANCH -f

        # Create the pull request
        gh pr create \
          --draft \
          --base ${{ inputs.base_branch }} \
          --head $RELEASE_BRANCH \
          --assignee ${{ inputs.assignee }} \
          --label "${{ inputs.label }}" \
          --title "Release - ${{ inputs.release_version }}" \
          --body-file changelog.txt \
```

Reply:
Thanks for spotting the issue.

Reply:
Fixed in https://github.com/nimblehq/github-actions-workflows/pull/7/commits/5cd6cc6c63389658f4a68d29b1c7c5169864e128